### ISSUE TITLE
Using Velero 1.8.1 with Restic, PV,PVC Data is not getting moved from Primary to DR Site, as we can see only PV and PVC for Stateful Application

### ISSUE THREAD

**Issue by prabhanjan060** [vmware-tanzu/velero#5018]
**What steps did you take and what happened:**
[A clear and concise description of what the bug is, and what commands you ran.)

> velero install     --provider aws     --plugins velero/velero-plugin-for-aws:v1.2.1     --bucket velero     --secret-file ./credentials-velero     --use-volume-snapshots=false  --backup-location-config region=minio,s3ForcePathStyle="true",s3Url=http://minio.velero.svc:9000/ --use-restic  --default-volumes-to-restic  --features=EnableCSI

Used tis command to take backup:

> velero backup create xxx-backup --selector app=xxx --include-namespaces velero-xxx  --default-volumes-to-restic

Setup info:

> 
> [rouser@rheldemo01 ~]$ kubectl get pods -n velero
> NAME                      READY   STATUS      RESTARTS   AGE
> minio-58fb7cfdc8-qr8qh    1/1     Running     0          2d23h
> minio-setup-lhxq2         0/1     Completed   0          2d23h
> restic-6l929              1/1     Running     0          47h
> velero-567dd85cfb-glnm8   1/1     Running     0          2d
> [rouser@rheldemo01` ~]$ kubectl get ns
> NAME                   STATUS   AGE
> default                Active   46d
> istio-system           Active   46d
> kube-node-lease        Active   46d
> kube-public            Active   46d
> kube-system            Active   46d
> kubernetes-dashboard   Active   46d
> nginx                  Active   42h
> nginx-example          Active   2d23h
> phpvelero              Active   47h
> velero                 Active   2d23h
> velero-php             Active   41h
> [rouser@rheldemo01 ~]$ kubectl get all -n velero-php
> NAME                      READY   STATUS    RESTARTS   AGE
> pod/php-5d84f86fb-4mp7l   1/1     Running   0          41h
> pod/php-mysql-0           1/1     Running   0          41h
> 
> NAME                TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)        AGE
> service/php         NodePort    10.97.193.220   <none>        80:30544/TCP   41h
> service/php-mysql   ClusterIP   None            <none>        3306/TCP       41h
> 
> NAME                            DESIRED   CURRENT   READY   AGE
> replicaset.apps/php-5d84f86fb   1         1         1       41h
> 
> NAME                         READY   AGE
> statefulset.apps/php-mysql   1/1     41h
> [rouser@rheldemo01 ~]$
> 

**What did you expect to happen:**
i am using this as learn page:
https://velero.io/docs/v1.9/contributions/minio/#docs

> $ kubectl -n velero get podvolumebackups -l velero.io/backup-name=mysql06-backup -o yaml
> apiVersion: v1
> items: []
> kind: List
> metadata:
>   resourceVersion: ""
>   selfLink: ""
> $

**The following information will help us better understand what's going [on**:](url)

_If you are using velero v1.7.0+:_   <Velero Version: 1.8.1>
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 

_If you are using earlier versions:_  
Please provide the output of the following commands (Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`): 
$ velero version
Client:
        Version: v1.8.1
        Git commit: 18ee078dffd9345df610e0ca9f61b31124e93f50
Server:
        Version: v1.8.1
[rouser@rheldemo01 ~]$
- Velero features (use `velero client config get features`): 
- [rouser@rheldemo01 ~]$ velero client config get features
features: <NOT SET>

- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version: 
-  kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6", GitCommit:"ad3338546da947756e8a88aa6822e9c11e7eac22", GitTreeState:"clean", BuildDate:"2022-04-14T08:49:13Z", GoVersion:"go1.17.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:19:12Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"linux/amd64"}
$ 
- Cloud provider or hardware configuration: MINKUBE
- OS (e.g. from `/etc/os-release`): 
[rouser@rheldemo01 ~]$ sudo cat /etc/os-release
NAME="Red Hat Enterprise Linux"
VERSION="8.5 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.5"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.5 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/8/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.5
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.5"
$

[bundle-2022-06-17-11-30-14_velero_backup_restore.tar.gz](https://github.com/vmware-tanzu/velero/files/8924983/bundle-2022-06-17-11-30-14_velero_backup_restore.tar.gz)
[bundle-2022-06-17-11-14-24_velero deug.tar.gz](https://github.com/vmware-tanzu/velero/files/8924986/bundle-2022-06-17-11-14-24_velero.deug.tar.gz)

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by blackpiglet** [comment]
@prabhanjan060 
Checked the bundle, I think this is related to resource filtering rules. There is no PodVolumeBackup created during backup process. It means Velero thinks no need to backup any Pod's volumes by backup setting.
You can get the resources included in the backup by command `velero backup describe xxx --details` to see whether the backup is created as your need.

```
            "spec": {
                "defaultVolumesToRestic": true,
                "hooks": {},
                "includedNamespaces": [
                    "velero-php"
                ],
                "labelSelector": {
                    "matchLabels": {
                        "app": "php"
                    }
                },
                "metadata": {},
                "storageLocation": "default",
                "ttl": "720h0m0s"
            }
```

I found most of the backups' spec look like this. To let the Restic work, backup needs to include the pod mounting the volumes. Please check whether the pods are labeled with "app": "php".

**Comment by prabhanjan060** [comment]
> [rouser@rheldemo01 ~]$ kubectl get pvc -n velero-php --show-labels
> NAME              STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE   LABELS
> portworx-sc-pvc   Bound    pvc-3dfdb89c-c02b-4dfc-a568-fa1989614fad   10Gi       RWO            standard       42h   app=php,velero.io/backup-name=mysql06-backup,velero.io/restore-name=mysql06-backup-20220615182353
> [rouser@rheldemo01 ~]$ kubectl get pods -n velero-php --show-labels
> NAME                  READY   STATUS    RESTARTS   AGE   LABELS
> php-5d84f86fb-4mp7l   1/1     Running   0          42h   app=php,pod-template-hash=5d84f86fb,tier=frontend,velero.io/backup-name=mysql06-backup,velero.io/restore-name=mysql06-backup-20220615182353
> php-mysql-0           1/1     Running   0          42h   app=php,controller-revision-hash=php-mysql-7ff98f7dd8,statefulset.kubernetes.io/pod-name=php-mysql-0,tier=mysql,velero.io/backup-name=mysql06-backup,velero.io/restore-name=mysql06-backup-20220615182353
> [rouser@rheldemo01 ~]$ kubectl get svc -n velero-php --show-labels
> NAME        TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)        AGE   LABELS
> php         NodePort    10.97.193.220   <none>        80:30544/TCP   42h   app=php,velero.io/backup-name=mysql06-backup,velero.io/restore-name=mysql06-backup-20220615182353
> php-mysql   ClusterIP   None            <none>        3306/TCP       42h   app=php,velero.io/backup-name=mysql06-backup,velero.io/restore-name=mysql06-backup-20220615182353
> [rouser@rheldemo01 ~]$


How to fix this issues:

[rouser@rheldemo01 ~]$ velero backup describe mysql06-backup --details
Name:         mysql06-backup
Namespace:    velero
Labels:       velero.io/storage-location=default
Annotations:  velero.io/source-cluster-k8s-gitversion=v1.23.3
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=23

Phase:  Completed

Errors:    0
Warnings:  1

Namespaces:
  Included:  velero-php
  Excluded:  <none>

Resources:
  Included:        *
  Excluded:        <none>
  Cluster-scoped:  auto

Label selector:  app=php

Storage Location:  default

Velero-Native Snapshot PVs:  auto

TTL:  720h0m0s

Hooks:  <none>

Backup Format Version:  1.1.0

Started:    2022-06-15 18:18:08 +0530 IST
Completed:  2022-06-15 18:18:10 +0530 IST

Expiration:  2022-07-15 18:18:08 +0530 IST

Total items to be backed up:  13
Items backed up:              13

Resource List:  <error getting backup resource list: Get "http://localhost:9000/velero/backups/mysql06-backup/mysql06-backup-resource-list.json.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20220617%2Fminio%2Fs3%2Faws4_request&X-Amz-Date=20220617T074431Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Signature=f2c412f8e3ec5dbef2955308e9c45eb4c31c98ae3c17afa1b60294a7d83082c3": dial tcp 127.0.0.1:9000: connect: connection refused>

Velero-Native Snapshots: <none included>
[rouser@rheldemo01 ~]$

**Comment by blackpiglet** [comment]
@prabhanjan060 
```
            "spec": {
                "config": {
                    "publicUrl": "http://localhost:9000",
                    "region": "minio",
                    "s3ForcePathStyle": "true",
                    "s3Url": "http://minio.velero.svc:9000"
                },
                "default": true,
                "objectStorage": {
                    "bucket": "velero"
                },
                "provider": "aws"
            },
```
Checked with your BSL configuration. I think need to update `publicUrl` as  `http://minio.velero.svc:9000`.
Then after port-forward, you need to add the `127.0.0.1 minio.velero.svc` into your /etc/hosts

**Comment by prabhanjan060** [comment]
[rouser@rheldemo01 ~]$ kubectl get backupstoragelocation default -n velero
NAME      PHASE       LAST VALIDATED   AGE    DEFAULT
default   Available   57s              3d1h   true
[rouser@rheldemo01 ~]$ kubectl edit backupstoragelocation default -n velero
apiVersion: velero.io/v1
kind: BackupStorageLocation
metadata:
  creationTimestamp: "2022-06-14T06:26:42Z"
  generation: 19
  labels:
    component: velero
  name: default
  namespace: velero
  resourceVersion: "2940691"
  uid: e71d5d4e-d988-4045-921f-3e32e171001d
spec:
  config:
    publicUrl: http://minio.velero.svc:9000
    region: minio
    s3ForcePathStyle: "true"
    s3Url: http://minio.velero.svc:9000
  default: true
  objectStorage:
    bucket: velero
  provider: aws
status:
  lastSyncedTime: "2022-06-17T08:17:19Z"
  lastValidationTime: "2022-06-17T08:17:20Z"
  phase: Available

[rouser@rheldemo01 ~]$ cat /etc/hosts
127.0.0.1   minio.velero.svc localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6

192.168.8.204 rheldemo01
[rouser@rheldemo01 ~]$

But the Curl is failing :

[rouser@rheldemo01 ~]$ curl http://localhost:9000/
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AccessDenied</Code><Message>Access Denied.</Message><Resource>/</Resource><RequestId>16F95A749C9A7D38</RequestId><HostId>28374de0-17e3-4a3a-b8a9-01caae8566c8</HostId></Error>[rouser@rheldemo01 ~]$

[rouser@rheldemo01 ~]$ curl http://minio.velero.svc:9000
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AccessDenied</Code><Message>Access Denied.</Message><Resource>/</Resource><RequestId>16F95A7A748CA7E4</RequestId><HostId>28374de0-17e3-4a3a-b8a9-01caae8566c8</HostId></Error>[rouser@rheldemo01 ~]$
[rouser@rheldemo01 ~]$

**Comment by blackpiglet** [comment]
During meeting, we resolved the MinIO access problem, and found the problem is related to Minikube's kubelet doesn't store pods related information under `/var/lib/kubelet/pods`. That causes Velero integrated Restic cannot read pod volumes from worker node host, and no PodVolumeBackups are created in backup.
Not sure about Minikube doesn't store pod information there by default, or that is your environment  specific.
I found something related to this topic. Looks like OC system setting is quite different. Maybe this is what your need.
https://velero.io/docs/v1.9/restic/#configure-restic-daemonset-spec
